server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -338,41 +369,18 @@ private void putUnderLock(BytesRef uid, VersionValue version, Maps maps) {
         assert uid.bytes.length == uid.length : "Oversized _uid! UID length: " + uid.length + ", bytes length: " + uid.bytes.length;
         if (version.isDelete() == false) {
             maps.put(uid, version);
-            removeTombstoneUnderLock(uid);
+            maps.tombstones.removeTombstone(uid);
         } else {
             DeleteVersionValue versionValue = (DeleteVersionValue) version;
-            putTombstone(uid, versionValue);
+            maps.tombstones.putTombstone(uid, versionValue);
             maps.remove(uid, versionValue);
         }
     }
 
-    private void putTombstone(BytesRef uid, DeleteVersionValue version) {
-        long uidRAMBytesUsed = BASE_BYTES_PER_BYTESREF + uid.bytes.length;
-        // Also enroll the delete into tombstones, and account for its RAM too:
-        final VersionValue prevTombstone = tombstones.put(uid, version);
-        long accountRam = (BASE_BYTES_PER_CHM_ENTRY + version.ramBytesUsed() + uidRAMBytesUsed);
-        // Deduct tombstones bytes used for the version we just removed or replaced:
-        if (prevTombstone != null) {
-            accountRam -= (BASE_BYTES_PER_CHM_ENTRY + prevTombstone.ramBytesUsed() + uidRAMBytesUsed);
-        }
-        if (accountRam != 0) {
-            long v = ramBytesUsedTombstones.addAndGet(accountRam);
-            assert v >= 0: "bytes=" + v;
-        }
-    }
-
-    /**
-     * Removes this uid from the pending deletes map.
-     */
+    // For testing
```

Comment:
This -- and the number of times we do `maps.tombstones.doStuff()` -- feels a little strange. What do you think about moving methods like this onto `Maps` instead?